code/parse/sexp.cpp

```diff
@@ -10539,7 +10546,7 @@ int eval_for_ship_collection(int arg_handler_node, int condition_node, int op_co
 				}
 
 				if (constraint_index == ship_index)
-					argument_vector.emplace_back(vm_strdup(shipp->ship_name), -1);
+					argument_vector.emplace_back(shipp->ship_name, -1);
```

Comment:
Why were these `vm_strdup()` calls removed?

Won't these changes result in attempts to free memory we shouldn't be trying to free?

Reply:
Hmm, yeah, that would be problematic.  Might need to figure out a way around that.